### ISSUE TITLE
Panel and Dialog components do not provide a role for header section

### ISSUE THREAD

**Issue by slumley1** [OfficeDev/office-ui-fabric-react#1309]
#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

#### Describe the issue:

The Panel and Dialog components are missing the accessibility role attribute for the header section.

#### Actual behavior:

Header element rendered as following:
`<p class="ms-Panel-headerText features-panel-header" id="Panel0-headerText">sample</p>`

#### Expected behavior:
Expect the role attribute to be set to heading:

`<p class="ms-Panel-headerText features-panel-header" id="Panel0-headerText" role="heading">sample</p>`
